src/SFML/Window/WindowBase.cpp

```diff
@@ -308,7 +308,7 @@ bool WindowBase::hasFocus() const
 ////////////////////////////////////////////////////////////
 WindowHandle WindowBase::getSystemHandle() const
 {
-    return m_impl ? m_impl->getSystemHandle() : 0;
+    return m_impl ? m_impl->getSystemHandle() : WindowHandle{};
```

Comment:
According to the [SFML doc](https://www.sfml-dev.org/documentation/2.5.1/group__window.php#gaed947028b0698a812cad2f97bfe9caa3), most handle types are pointers (`HWND` is one, not sure about X11 `Window`), so I wonder if we shouldn't use `nullptr` here?

`WindowHandle{}` is more generic, but could possibly also represent a type which cannot be explicitly checked for a "null state". So if we lose that ability (and the user cannot check the returned handle for validity), I'd probably prefer a compile error.

Reply:
Force-pushing is the culprit here, but originally I had changed this to 

```cpp
return m_impl ? m_impl->getSystemHandle() : nullptr;
```

which caused a compilation failure on Unix, because we have

```cpp
#elif defined(SFML_SYSTEM_LINUX) || defined(SFML_SYSTEM_FREEBSD) || defined(SFML_SYSTEM_OPENBSD) || defined(SFML_SYSTEM_NETBSD)

    // Window handle is Window (unsigned long) on Unix - X11
    using WindowHandle = unsigned long;
```

So using `WindowHandle{}` is the best choice, IMHO.

Reply:
I see, thanks for explanation!